add `status` command and chaining. (#122)
OpenZeppelin/nile: README.md

Compile Cairo contracts. Compilation artifacts are written into the `artifacts/` directory.

```txt

> nile compile [PATH_TO_CONTRACT | --directory DIRECTORY] [--disable-hint-validation]

Reviewer: ```suggestion
nile compile [PATH_TO_CONTRACT] [--directory DIRECTORY] [--disable-hint-validation]
```
`PATH_TO_CONTRACT` and `--directory` are not really alternatives aren't they?

Reviewer 2: To my eyes, when you add `PATH_TO_CONTRACT`, `--directory` serves no purpose and vice versa, no?

`nile compile path/to/contract.cairo`

and

`nile compile --directory path/to/directory`

Reviewer: I believe it also sets the cairo_path for the compiler. for example, this is needed to compile OZ contracts:
```sh
nile compile src/openzeppelin/account/presets/EthAccount.cairo --directory src
```

> NOTICE: this method doesn't use an account, which will be deprecated very soon as StarkNet makes deployments from accounts mandatory.

```sh

> nile deploy contract --alias my_contract [TRACK/DEBUG]

Reviewer: ```suggestion
nile deploy contract --alias my_contract [--track/--debug]
```
i'm not sure i'm familiar with that syntax, shouldn't it be like this?

A few things to notice here:

- `nile deploy <contract_name>` looks for an artifact with the same name.

> - This created a `localhost.deployments.txt` file storing all data related to my deployment.

Reviewer 3: We should maintain consistency in grammar for the README: 

```suggestion
- This creates or updates the` localhost.deployments.txt` file storing all data related to deployments.
```

There are more places where this happens, like the next sentence, where instead of "lets me create" I would say something like "allows creating".

I think we should maintain third-person singular, but if not, at least we should be consistent.

I would prioritize an issue for refactoring deeply the README for this, or the next milestone @martriay @andrew-fleming .

Reviewer 2: Mmm yeah, I'm not a fan of the first-person. I'd say third-person singular or even second person ("lets you create"). I agree on prioritizing a README refactor.

Reviewer 3: I saw now I wrote third-person singular where I meant third-person present, sorry.

Reviewer 2: Haha I rewrote the same thing. I understood what you meant :)

> A few things to notice here:

Reviewer: ```suggestion
A few things to note here:
```
this is better right?

- `nile deploy <contract_name>` looks for an artifact with the same name.
- This creates or updates the` localhost.deployments.txt` file storing all data related to deployments.
- The `--alias` parameter creates a unique identifier for future interactions, if no alias is set then the contract's address can be used as identifier.
- By default Nile works on local, but you can use the `--network` parameter to interact with `mainnet`, `goerli`, and the default `localhost`.
- By default, the ABI corresponding to the contract will be registered with the deployment. To register a different ABI file, use the `--abi` parameter.

> - `--track` and `--debug` flags can be used to chain the `setup` call with `nile status` + the chosen flag. See `status` below for a complete description.

Reviewer: ```suggestion
- `--track` and `--debug` flags can be used to watch the status of the deployment transaction. See `status` below for a complete description.
```

A few things to note here:

- `nile setup <private_key_alias>` looks for an environment variable with the name of the private key alias.
- This creates or updates `localhost.accounts.json` file storing all data related to accounts management.
- The creates or updates `localhost.deployments.txt` file storing all data related to deployments.

> - `--track` and `--debug` flags can be used to chain the `setup` call with `nile status` + the chosen flag. See `status` below for a complete description.

Reviewer: ```suggestion
- `--track` and `--debug` flags can be used to watch the status of the account deployment transaction. See `status` below for a complete description.
```

A few things to note here:

- `nile setup <private_key_alias>` looks for an environment variable with the name of the private key alias.
- This creates or updates `localhost.accounts.json` file storing all data related to accounts management.
- The creates or updates `localhost.deployments.txt` file storing all data related to deployments.
- `--track` and `--debug` flags can be used to watch the status of the account deployment transaction. See `status` below for a complete description.

### `send`

Execute a transaction through the `Account` associated with the private key provided. The syntax is:

```sh

> nile send [--track | --debug] <private_key_alias> <contract_identifier> <method> [PARAM_1, PARAM2...]

Reviewer: ```suggestion
nile send <private_key_alias> <contract_identifier> <method> [PARAM_1, PARAM2...] [--track | --debug]
```

Some things to note:

- This sends the transaction to the network by default, but you can use the `--estimate_fee` flag to estimate the fee without sending the transaction, or the `--simulate` flag to get a traceback of the simulated execution.
- `max_fee` defaults to `0`. Add `--max_fee <max_fee>` to set the maximum fee for the transaction.
- `network` defaults to the `localhost`. Add `--network <network>` to change the network for the transaction.

> - `--track` and `--debug` flags can be used to chain the `send` call with `nile status` + the chosen flag. See `status` below for a complete description.

Reviewer: ```suggestion
- `--track` and `--debug` flags can be used to watch the status of the transaction. See `status` below for a complete description.
```

### `declare`

Very similar to `send`, but for declaring a contract based on its name through an account.

```sh
nile declare <private_key_alias> contract --alias my_contract

🚀 Declaring contract
⏳ Successfully sent declaration of contract as 0x07ec10eb0758f7b1bc5aed0d5b4d30db0ab3c087eba85d60858be46c1a5e4680
🧾 Transaction hash: 0x7222604b048632326f6a016ccb16fbdea7e926cd9e2354544800667a970aee4
📦 Registering declaration as my_contract in localhost.declarations.txt
```

A few things to notice here:

- `nile declare <private_key_alias> <contract_name>` looks for an artifact with name `<contract_name>`.
- This creates or updates a `localhost.declarations.txt` file storing all data related to your declarations.
- The `--alias` parameter lets you create a unique identifier for future interactions, if no alias is set then the contract's address can be used as identifier.
- By default Nile works on local, but you can use the `--network` parameter to interact with `mainnet`, `goerli`, and the default `localhost`.

> - `--track` and `--debug` flags can be used to chain the `declare` call with `nile status` + the chosen flag. See `status` below for a complete description.

Reviewer: ```suggestion
- `--track` and `--debug` flags can be used to watch the status of the declaration transaction. See `status` below for a complete description.
```

Prints the current status of a transaction.

```sh

> nile status <transaction_hash> [CONTRACTS_FILE, NETWORK, --track, --debug]

Reviewer: i think this means you can use both together

nile version
```

### `status`

Prints the current status of a transaction.

```sh

> nile status <transaction_hash> [CONTRACTS_FILE, NETWORK, TRACK/DEBUG]

Reviewer: ```suggestion
nile status <transaction_hash> [CONTRACTS_FILE, NETWORK, TRACK/DEBUG]
```
these examples are inconsistent with the recent changes (`--track/--debug`). i'm still a bit confused by this syntax. is it common to refer to flags like this?

Reviewer: because i see it used for other flags/params as well

Reviewer 2: I'm honestly not sure if it's common. I'll update the examples to something less confusing

nile version
```

### `status`

Prints the current status of a transaction.

```sh
nile status <transaction_hash> [CONTRACTS_FILE, NETWORK, TRACK, DEBUG]
⏳ Querying the network for transaction status...
✅ Transaction status: ACCEPTED ON L1. No error in transaction.
```

> #### `--track`, `-t` flag

Reviewer 3: Linking the same comment I made for a similar following comment:
https://github.com/OpenZeppelin/nile/pull/122#discussion_r1002593840

> #### `--track`, `-t` flag

Reviewer: i can understand you don't want to block the terminal after sending a transaction, but when explicitly requesting the status you want to know whether a tx is confirmed or rejected, i expect the primary use case therefore to be waiting for a resolution when status is pending. if you want to continue, yo can `ctrl + c`

In case of pending transaction states, continue probing the network. Here in the case of a successful transaction.

```sh
nile status -t <transaction_hash> [CONTRACTS_FILE, NETWORK, DEBUG]
⏳ Querying the network for transaction status...
🕒 Transaction status: NOT_RECEIVED. Trying again in a moment...
🕒 Transaction status: RECEIVED. Trying again in a moment...
🕒 Transaction status: PENDING. Trying again in a moment...
✅ Transaction status: ACCEPTED_ON_L2. No error in transaction.
```

#### `--debug`, `-d` flag

Use locally available contracts to make error messages from rejected transactions more explicit.  

> NB: Implies `--track`.

Reviewer 2: Nota Bene (Latin for "note well"). I changed it to "Note"
